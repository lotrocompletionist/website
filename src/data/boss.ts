export interface IBoss {
  id: number;
  name: string;
}
