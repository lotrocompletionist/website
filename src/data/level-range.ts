export interface ILevelRange {
  minimum: number;
  maximum?: number;
}
