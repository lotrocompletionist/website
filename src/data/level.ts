export interface Level {
  minimum: number;
  maximum?: number;
}
