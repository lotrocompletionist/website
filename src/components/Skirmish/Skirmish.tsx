import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { Level } from "../Level/Level";
import { ISkirmish } from "../../data/skirmish";

export interface SkirmishProps {
  skirmish: ISkirmish;
}

export function Skirmish(props: SkirmishProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.skirmish.name}</Title>
          </MediaContent>
        </Media>
        <Content>
          {props.skirmish.type}
          <br />
          <small>
            <Level level={props.skirmish.level} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
