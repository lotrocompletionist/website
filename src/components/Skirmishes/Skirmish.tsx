import * as React from "react";
import {
  CardContent,
  Media,
  MediaLeft,
  MediaContent,
  Content,
  Image,
  Title,
  Subtitle
} from "bloomer";
import { LevelRange } from "@app/components/Shared/LevelRange/LevelRange";
import { Bosses } from "@app/components/Shared/Bosses/Bosses";
import { ISkirmish } from "@app/data/skirmish";

export interface SkirmishProps {
  skirmish: ISkirmish;
}

export function Skirmish(props: SkirmishProps) {
  return (
    <>
      <CardContent>
        <Media>
          <MediaLeft>
            <Image isSize="48x48" src="https://via.placeholder.com/96x96" />
          </MediaLeft>
          <MediaContent>
            <Title isSize={4}>{props.skirmish.name}</Title>
          </MediaContent>
        </Media>
        <Content>
          {props.skirmish.type}
          <br />
          <small>
            <LevelRange range={props.skirmish.level} />
            <br />
            <Bosses bosses={props.skirmish.bosses} />
          </small>
        </Content>
      </CardContent>
    </>
  );
}
