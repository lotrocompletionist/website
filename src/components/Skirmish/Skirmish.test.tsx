import * as React from "react";
import { create } from "react-test-renderer";
import { Skirmish } from "./Skirmish";
import { ISkirmish } from "../../data/skirmish";

test("<Skirmish> renders correctly", () => {
  const skirmish: ISkirmish = {
    id: 1,
    name: "The Siege of Gondamon",
    level: { minimum: 20, maximum: 115 },
    faction: "Thorin's Hall",
    type: "Defense",
    requirements: ["None"],
    bosses: [
      { id: 1, name: "Thrastar" },
      { id: 2, name: "Hergill" },
      { id: 3, name: "Agi Gatebreaker" },
      { id: 4, name: "Ghashn√¢kh" },
      { id: 5, name: "Sigurmar" }
    ]
  };
  const tree = create(<Skirmish skirmish={skirmish} />).toJSON();
  expect(tree).toMatchSnapshot();
});
