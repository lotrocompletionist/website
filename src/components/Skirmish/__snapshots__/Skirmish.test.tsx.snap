// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skirmish> renders correctly 1`] = `
<div
  className="card-content"
>
  <article
    className="media"
  >
    <div
      className="media-left"
    >
      <figure
        className="image is-48x48"
      >
        <img
          src="https://via.placeholder.com/96x96"
        />
      </figure>
    </div>
    <div
      className="media-content"
    >
      <h1
        className="title is-4"
      >
        The Siege of Gondamon
      </h1>
    </div>
  </article>
  <div
    className="content"
  >
    Defense
    <br />
    <small>
      Level: 
      20
       - 115
    </small>
  </div>
</div>
`;
